aws/glue/crawlers.tf

```diff
@@ -0,0 +1,113 @@
+resource "aws_glue_security_configuration" "encryption_at_rest" {
+  name = "encryption-at-rest"
+
+  encryption_configuration {
+    cloudwatch_encryption {
+      cloudwatch_encryption_mode = "SSE-KMS"
+      kms_key_arn                = aws_kms_key.aws_glue.arn
+    }
+
+    job_bookmarks_encryption {
+      job_bookmarks_encryption_mode = "CSE-KMS"
+      kms_key_arn                   = aws_kms_key.aws_glue.arn
+    }
+
+    s3_encryption {
+      s3_encryption_mode = "SSE-S3"
+    }
+  }
+}
+
+#
+# RDS Usage Report
+#
+resource "aws_glue_crawler" "forms_rds_data" {
+  name          = "RDS Usage Report"
+  description   = "Classify the Forms RDS data"
+  database_name = "rds_db_catalog"
+  table_prefix  = "rds_report_"
+
+  role                   = aws_iam_role.glue_crawler.arn
+  security_configuration = aws_glue_security_configuration.encryption_at_rest.name
+
+  s3_target {
+    path = "s3://${var.datalake_bucket_name}/processed-data/template/"
+  }
+
+  s3_target {
+    path = "s3://${var.datalake_bucket_name}/processed-data/user/"
+  }
+
+  s3_target {
+    path = "s3://${var.datalake_bucket_name}/processed-data/templateToUser/"
+  }
+
+  configuration = jsonencode(
+    {
+      CrawlerOutput = {
+        Tables = {
+          TableThreshold = 3
+        }
+      }
+      CreatePartitionIndex = true
+      Version              = 1
+  })
+
+  schedule = "cron(00 7 1 * ? *)" # Create the new month's partition key
```

Comment:
Question about your data here.  Will you ever have a scenario where it's possible that the new month partition doesn't always exist on the first day of the month?  

We've got this case in a few of our dataset's so we need to run the crawler for the first few days of each month just to make sure we catch the month partition appearing.  Also totally understand if this is not the case for you and running on day 1 is enough.

Reply:
I picked an arbitrary date. :) 

@srtalbot - we need to come to a consensus on how often we pull data from the app. Do we want it daily? Weekly? Monthly? Some other unit of time? What would be of most value?